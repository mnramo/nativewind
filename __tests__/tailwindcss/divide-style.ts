import { ViewStyle } from "react-native";
import { createTests, expectError, tailwindRunner } from "./runner";

const scenarios: Record<string, ViewStyle["borderStyle"]> = {
  solid: "solid",
  dashed: "dashed",
  dotted: "dotted",
};

tailwindRunner("Border - Divide Style", [
  ...createTests("divide", scenarios, (n) => ({
    media: ["--general-sibling-combinator"],
    style: {
      borderStyle: n,
    },
  })),
  ...expectError(["divide-double", "divide-none"]),
]);
