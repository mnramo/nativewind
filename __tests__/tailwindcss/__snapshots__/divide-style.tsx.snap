// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Border - Divide Style divide-dashed 1`] = `
<View
  style={Array []}
>
  <Text>
    A
  </Text>
  <Text
    style={
      Array [
        Object {
          "borderStyle": "dashed",
        },
      ]
    }
  >
    B
  </Text>
</View>
`;

exports[`Border - Divide Style divide-dotted 1`] = `
<View
  style={Array []}
>
  <Text>
    A
  </Text>
  <Text
    style={
      Array [
        Object {
          "borderStyle": "dotted",
        },
      ]
    }
  >
    B
  </Text>
</View>
`;

exports[`Border - Divide Style divide-solid 1`] = `
<View
  style={Array []}
>
  <Text>
    A
  </Text>
  <Text
    style={
      Array [
        Object {
          "borderStyle": "solid",
        },
      ]
    }
  >
    B
  </Text>
</View>
`;
